core/src/test/resources/sql/sub-query.iq

```diff
@@ -464,7 +464,7 @@ EnumerableCalc(expr#0..2=[{inputs}], proj#0..1=[{exprs}])
 with t (a, b) as (select * from (values (60, 'b')))
 select * from t where a in (select deptno from "scott".dept);
 EnumerableCalc(expr#0..2=[{inputs}], A=[$t1], B=[$t2])
-  EnumerableMergeJoin(condition=[=($0, $1)], joinType=[inner])
+  EnumerableMergeJoin(condition=[=($1, $0)], joinType=[inner])
```

Comment:
Any idea what caused this change?

Reply:
Ping @jinxing64

Reply:
Thanks for ping @hsyuan 
I'm super busy these days and sorry for late reply, I will resolve comments today ~

Reply:
Logical plan of
 ```
with t (a, b) as (select * from (values (60, 'b'))) 
select * from t where a in (select deptno from "scott".dept);
```
is as below
```
LogicalProject(A=[$0], B=[$1])
  LogicalJoin(condition=[=($0, $2)], joinType=[inner])
    LogicalProject(EXPR$0=[$0], EXPR$1=[$1])
      LogicalValues(tuples=[[{ 60, 'b' }]])
    LogicalProject(DEPTNO=[$0])
      EnumerableTableScan(table=[[scott, DEPT]])
```
So we can easy understand that when JoinCommuteRule exchanges left & right child, the join condition will be reversed accordingly;
Before this PR, `JoinInfo` will help sort out of the equi-condition (https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableMergeJoinRule.java#L97)
After this change, the join condition is passed in EnumerableMergeJoin directly. Thus the change happens